### DISCUSSION TITLE
ASA sorted by absolute values

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#66] [General]
We had got [an issue](https://github.com/asastats/channel/issues/64) with negative/borrowed values affecting the ASA chart and we have solved it by dismissing negative values completely from the calculation of the percentages.

And that issue is a motivation for the next discussion: should we sort ASA by absolute values rather than values?

Right now those borrowed amounts are shown below all other ASA, while an alternative is to sort them like:

```
 2,543.25 ALGO
-1,200.43 ALGO
   754.29 ALGO
    42.88 ALGO
   -18.91 ALGO
     2.12 ALGO

```

**Comment by wxivd** [comment]
> should we sort ASA by absolute values rather than values?

We can argue both ways. Absolute value highlights the largest amount regardless of (owned/borrowed) which is as important as sorting values. 

How feasible it is to provide these two options as a feature?

**Reply by ipaleka** [reply]
> How feasible it is to provide these two options as a feature?

Well, the features like this one end up in https://github.com/asastats/channel/wiki/FeaturesDependentOnUserSettings . Implementing this inside the user settings system would be easier.

**Reply by wxivd** [reply]
> Well, the features like this one end up in https://github.com/asastats/channel/wiki/FeaturesDependentOnUserSettings . Implementing this inside the user settings system would be easier.

[Done](https://github.com/asastats/channel/wiki/FeaturesDependentOnUserSettings), Thank you

**Reply by ipaleka** [reply]
And now our discussion continues with the question: what would be the default view for our users? :)

**Reply by wxivd** [reply]
[Characteristic-Sorted Portfolios: Estimation and Inference](https://direct.mit.edu/rest/article/102/3/531/96794/Characteristic-Sorted-Portfolios-Estimation-and). [PDF Document](https://www.newyorkfed.org/medialibrary/media/research/staff_reports/sr788.pdf?la=en):
- The premise behind portfolio sorting is to discover whether expected returns of an asset are related to a certain characteristic. Therefore, technical discussion may be important at the stage of structuring the development of **advanced feature for Return on investment (estimation, inference)**. 
- Portfolio sorting tools: used to display asset pricing, highlight pricing anomalies, identify profitable investment strategies. 
- The paper discuss the statistical properties of the procedure and propose a framework for sorting.



> what would be the default view for our users? 

Now, from non-technical point of view

1- Sorting by Values, serves the classification and distinction between different types of assets (Owned, Borrowed). A matter of fact, a separate pie Chart for borrowed assets may be utilised as a feature in future once lending/borrowing DeFi sector flourishes to include more than one type of assets.
2- Absolute value is closely related to the notion of magnitude in various mathematical and physical contacts. In finance, I think the direction / vector of value is more important than the magnitude of value.
3- The way we display assets builds and shapes our UX. If we plan to distinguish between owned and borrowed assets in advanced widgets in the future. Then it is logical to maintain the current sorting as the default view.

**Reply by ipaleka** [reply]
> 1- Sorting by Values, serves the classification and distinction between different types of assets (Owned, Borrowed). A matter of fact, a separate pie Chart for borrowed assets may be utilised as a feature in future once lending/borrowing DeFi sector flourishes to include more than one type of assets. 

Hm, this is interesting thought... We have had that bug recently with percentages in charts that caused slices to become "others" because we hide in chart assets having less than 5% of the total value. So the solution for now is to just truncate negative values from the total and calculate percentages from the total consisting only of positive values.

That whole thing with negative values is complicated to present intuitively and also *breaks* math sometimes. 

I guess we'd be fine with that approach in the future, but we are ready to implement any other idea if it is meaningful.


> 2- Absolute value is closely related to the notion of magnitude in various mathematical and physical contacts. In finance, I think the direction / vector of value is more important than the magnitude of value. 3- The way we display assets builds and shapes our UX. If we plan to distinguish between owned and borrowed assets in advanced widgets in the future. Then it is logical to maintain the current sorting as the default view.

Well, our approach has been to group all things related to an ASA under one roof. 

That being said, it would be kind of closer to having the corresponding chart slice separated into two regions: owned vs. borrowed. So let's say the slice is fifth in the portfolio and so it has got orange color. If the account holds 520 of that ASA and has borrowed 130 of it (meaning that 390 is used in calculation to get that fifth place), then its slice will be presented as a piece where 75% is colored in that orange color, while 25% is colored by its complementary color (blue).
